### DISCUSSION TITLE
What is the meaning of the keyword `event`?

### DISCUSSION THREAD

**Discussion by ghost** [pascalabcnet/pascalabcnet#3017] [Q&A] [answered]
It's listed on `Идентификаторы и ключевые слова` of `Структура программы`. But I can't find anything about it on the rest of the help file.

Update: So many keywords are undocumented, some of them are completely new and specific to PascalABC.NET that a Google search with `keyword + delphi` doesn't give any relevant results.

**Comment by SunSerega** [comment, marked as answer]
`event` is a .Net thing. You'll probably have more luck searching for `keyword+C#` instead.

It's a field modifier, that only works if the field type is a delegate.
It allows things outside of the class to only `+=` and `-=` handlers for that field.
While inside of the class that defined `event`-field you can also invoke all those handlers.

I think you've already seen events from outside:
```
window.Closing += (o,e) -> ...
```
Here `.Closing` is an event.
